apps/emqx/src/emqx_persistent_session_ds.erl

```diff
@@ -1029,75 +889,230 @@ receive_maximum(ConnInfo) ->
 expiry_interval(ConnInfo) ->
     maps:get(expiry_interval, ConnInfo, 0).
 
+bump_interval() ->
+    emqx_config:get([session_persistence, last_alive_update_interval]).
+
+%%--------------------------------------------------------------------
+%% SeqNo tracking
+%% --------------------------------------------------------------------
+
+-spec update_seqno(puback | pubrec | pubcomp, emqx_types:packet_id(), session()) ->
+    {ok, emqx_types:message(), session()} | {error, _}.
+update_seqno(Track, PacketId, Session = #{id := SessionId, s := S}) ->
+    SeqNo = packet_id_to_seqno(PacketId, S),
```

Comment:
Q: Should we reject things like `PUBREC(PacketID = 42)` (i.e. packet id from different track)?